// @tamagui/colors/dist/cjs/light/bronze.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var bronze_exports = {};
__export(bronze_exports, {
  bronze: () => bronze
});
module.exports = __toCommonJS(bronze_exports);
const bronze = {
  bronze1: "hsl(15, 30.0%, 99.1%)",
  bronze2: "hsl(17, 63.6%, 97.8%)",
  bronze3: "hsl(17, 42.1%, 95.2%)",
  bronze4: "hsl(17, 35.2%, 92.1%)",
  bronze5: "hsl(17, 31.5%, 88.2%)",
  bronze6: "hsl(17, 29.0%, 83.0%)",
  bronze7: "hsl(17, 26.9%, 75.6%)",
  bronze8: "hsl(17, 25.1%, 66.5%)",
  bronze9: "hsl(17, 20.0%, 54.0%)",
  bronze10: "hsl(17, 18.1%, 50.1%)",
  bronze11: "hsl(15, 20.0%, 43.1%)",
  bronze12: "hsl(12, 22.0%, 21.5%)"
};
