// @tamagui/colors/dist/cjs/light/bronzeA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var bronzeA_exports = {};
__export(bronzeA_exports, {
  bronzeA: () => bronzeA
});
module.exports = __toCommonJS(bronzeA_exports);
const bronzeA = {
  bronzeA1: "hsla(0, 89.3%, 18.3%, 0.012)",
  bronzeA2: "hsla(17, 95.1%, 40.1%, 0.036)",
  bronzeA3: "hsla(18, 98.3%, 29.8%, 0.067)",
  bronzeA4: "hsla(17, 99.6%, 26.0%, 0.106)",
  bronzeA5: "hsla(19, 99.6%, 23.8%, 0.157)",
  bronzeA6: "hsla(17, 99.2%, 22.5%, 0.220)",
  bronzeA7: "hsla(18, 99.7%, 21.6%, 0.310)",
  bronzeA8: "hsla(17, 99.5%, 20.2%, 0.420)",
  bronzeA9: "hsla(18, 99.9%, 16.7%, 0.553)",
  bronzeA10: "hsla(17, 99.2%, 15.4%, 0.589)",
  bronzeA11: "hsla(15, 99.9%, 13.2%, 0.655)",
  bronzeA12: "hsla(12, 98.7%, 5.7%, 0.832)"
};
