// @tamagui/colors/dist/cjs/light/brown.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var brown_exports = {};
__export(brown_exports, {
  brown: () => brown
});
module.exports = __toCommonJS(brown_exports);
const brown = {
  brown1: "hsl(30, 40.0%, 99.1%)",
  brown2: "hsl(30, 50.0%, 97.6%)",
  brown3: "hsl(30, 52.5%, 94.6%)",
  brown4: "hsl(30, 53.0%, 91.2%)",
  brown5: "hsl(29, 52.9%, 86.8%)",
  brown6: "hsl(29, 52.5%, 80.9%)",
  brown7: "hsl(29, 51.5%, 72.8%)",
  brown8: "hsl(28, 50.0%, 63.1%)",
  brown9: "hsl(28, 34.0%, 51.0%)",
  brown10: "hsl(27, 31.8%, 47.6%)",
  brown11: "hsl(25, 30.0%, 41.0%)",
  brown12: "hsl(20, 30.0%, 19.0%)"
};
