// @tamagui/colors/dist/cjs/light/crimsonA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var crimsonA_exports = {};
__export(crimsonA_exports, {
  crimsonA: () => crimsonA
});
module.exports = __toCommonJS(crimsonA_exports);
const crimsonA = {
  crimsonA1: "hsla(340, 100%, 51.0%, 0.012)",
  crimsonA2: "hsla(330, 100%, 51.0%, 0.032)",
  crimsonA3: "hsla(332, 99.1%, 47.1%, 0.063)",
  crimsonA4: "hsla(331, 99.9%, 44.3%, 0.102)",
  crimsonA5: "hsla(333, 99.9%, 42.3%, 0.153)",
  crimsonA6: "hsla(333, 99.5%, 40.5%, 0.224)",
  crimsonA7: "hsla(335, 99.7%, 39.1%, 0.322)",
  crimsonA8: "hsla(336, 99.5%, 38.5%, 0.440)",
  crimsonA9: "hsla(336, 99.9%, 44.3%, 0.761)",
  crimsonA10: "hsla(336, 100%, 42.5%, 0.808)",
  crimsonA11: "hsla(336, 99.8%, 40.3%, 0.883)",
  crimsonA12: "hsla(340, 99.0%, 10.0%, 0.950)"
};
