// @tamagui/colors/dist/cjs/light/gold.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var gold_exports = {};
__export(gold_exports, {
  gold: () => gold
});
module.exports = __toCommonJS(gold_exports);
const gold = {
  gold1: "hsl(50, 20.0%, 99.1%)",
  gold2: "hsl(47, 52.9%, 96.7%)",
  gold3: "hsl(46, 38.2%, 93.7%)",
  gold4: "hsl(44, 32.7%, 90.1%)",
  gold5: "hsl(43, 29.9%, 85.7%)",
  gold6: "hsl(41, 28.3%, 79.8%)",
  gold7: "hsl(39, 27.6%, 71.9%)",
  gold8: "hsl(36, 27.2%, 61.8%)",
  gold9: "hsl(36, 20.0%, 49.5%)",
  gold10: "hsl(36, 19.8%, 45.7%)",
  gold11: "hsl(36, 20.0%, 39.0%)",
  gold12: "hsl(36, 16.0%, 20.0%)"
};
