// @tamagui/colors/dist/cjs/light/goldA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var goldA_exports = {};
__export(goldA_exports, {
  goldA: () => goldA
});
module.exports = __toCommonJS(goldA_exports);
const goldA = {
  goldA1: "hsla(60, 89.3%, 18.3%, 0.012)",
  goldA2: "hsla(47, 99.9%, 34.6%, 0.051)",
  goldA3: "hsla(45, 97.0%, 27.9%, 0.087)",
  goldA4: "hsla(46, 98.0%, 25.4%, 0.134)",
  goldA5: "hsla(43, 98.4%, 22.6%, 0.185)",
  goldA6: "hsla(41, 99.7%, 22.0%, 0.259)",
  goldA7: "hsla(38, 99.8%, 21.5%, 0.357)",
  goldA8: "hsla(36, 99.3%, 21.5%, 0.487)",
  goldA9: "hsla(36, 99.9%, 16.2%, 0.604)",
  goldA10: "hsla(36, 99.2%, 14.6%, 0.636)",
  goldA11: "hsla(35, 99.1%, 11.2%, 0.687)",
  goldA12: "hsla(38, 98.0%, 3.8%, 0.832)"
};
