// @tamagui/colors/dist/cjs/light/grass.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var grass_exports = {};
__export(grass_exports, {
  grass: () => grass
});
module.exports = __toCommonJS(grass_exports);
const grass = {
  grass1: "hsl(116, 50.0%, 98.9%)",
  grass2: "hsl(120, 60.0%, 97.1%)",
  grass3: "hsl(120, 53.6%, 94.8%)",
  grass4: "hsl(121, 47.5%, 91.4%)",
  grass5: "hsl(122, 42.6%, 86.5%)",
  grass6: "hsl(124, 39.0%, 79.7%)",
  grass7: "hsl(126, 37.1%, 70.2%)",
  grass8: "hsl(131, 38.1%, 56.3%)",
  grass9: "hsl(131, 41.0%, 46.5%)",
  grass10: "hsl(132, 43.1%, 42.2%)",
  grass11: "hsl(133, 50.0%, 32.5%)",
  grass12: "hsl(130, 30.0%, 14.9%)"
};
