// @tamagui/colors/dist/cjs/light/grassA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var grassA_exports = {};
__export(grassA_exports, {
  grassA: () => grassA
});
module.exports = __toCommonJS(grassA_exports);
const grassA = {
  grassA1: "hsla(120, 94.9%, 38.7%, 0.016)",
  grassA2: "hsla(120, 94.9%, 38.7%, 0.048)",
  grassA3: "hsla(120, 98.0%, 35.5%, 0.079)",
  grassA4: "hsla(120, 98.7%, 31.5%, 0.126)",
  grassA5: "hsla(122, 98.5%, 29.9%, 0.193)",
  grassA6: "hsla(125, 99.2%, 27.9%, 0.283)",
  grassA7: "hsla(125, 99.9%, 27.0%, 0.408)",
  grassA8: "hsla(131, 100%, 27.6%, 0.604)",
  grassA9: "hsla(131, 99.7%, 26.3%, 0.726)",
  grassA10: "hsla(132, 99.9%, 24.0%, 0.761)",
  grassA11: "hsla(133, 99.5%, 19.5%, 0.840)",
  grassA12: "hsla(128, 98.0%, 4.9%, 0.895)"
};
