// @tamagui/colors/dist/cjs/light/greenA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var greenA_exports = {};
__export(greenA_exports, {
  greenA: () => greenA
});
module.exports = __toCommonJS(greenA_exports);
const greenA = {
  greenA1: "hsla(140, 94.9%, 38.7%, 0.016)",
  greenA2: "hsla(138, 99.9%, 38.5%, 0.051)",
  greenA3: "hsla(139, 97.7%, 36.9%, 0.087)",
  greenA4: "hsla(139, 98.5%, 32.7%, 0.134)",
  greenA5: "hsla(141, 100%, 30.4%, 0.200)",
  greenA6: "hsla(142, 99.0%, 28.9%, 0.295)",
  greenA7: "hsla(146, 99.5%, 27.6%, 0.428)",
  greenA8: "hsla(151, 99.5%, 28.8%, 0.644)",
  greenA9: "hsla(151, 99.9%, 28.0%, 0.812)",
  greenA10: "hsla(152, 99.6%, 25.8%, 0.840)",
  greenA11: "hsla(153, 99.9%, 21.0%, 0.906)",
  greenA12: "hsla(155, 99.4%, 6.2%, 0.918)"
};
