// @tamagui/colors/dist/cjs/light/indigoA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var indigoA_exports = {};
__export(indigoA_exports, {
  indigoA: () => indigoA
});
module.exports = __toCommonJS(indigoA_exports);
const indigoA = {
  indigoA1: "hsla(240, 92.6%, 26.5%, 0.008)",
  indigoA2: "hsla(223, 100%, 51.0%, 0.028)",
  indigoA3: "hsla(224, 100%, 50.1%, 0.059)",
  indigoA4: "hsla(223, 98.0%, 48.5%, 0.099)",
  indigoA5: "hsla(225, 98.6%, 46.4%, 0.150)",
  indigoA6: "hsla(224, 99.5%, 44.9%, 0.224)",
  indigoA7: "hsla(225, 99.7%, 43.9%, 0.318)",
  indigoA8: "hsla(226, 99.5%, 43.1%, 0.448)",
  indigoA9: "hsla(226, 100%, 41.2%, 0.757)",
  indigoA10: "hsla(226, 99.8%, 37.1%, 0.773)",
  indigoA11: "hsla(226, 99.6%, 31.1%, 0.797)",
  indigoA12: "hsla(226, 99.3%, 11.4%, 0.938)"
};
