// @tamagui/colors/dist/cjs/light/limeA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var limeA_exports = {};
__export(limeA_exports, {
  limeA: () => limeA
});
module.exports = __toCommonJS(limeA_exports);
const limeA = {
  limeA1: "hsla(80, 93.8%, 31.4%, 0.020)",
  limeA2: "hsla(85, 99.3%, 40.2%, 0.059)",
  limeA3: "hsla(84, 98.7%, 43.2%, 0.138)",
  limeA4: "hsla(84, 99.6%, 43.0%, 0.220)",
  limeA5: "hsla(85, 99.8%, 41.8%, 0.310)",
  limeA6: "hsla(82, 99.8%, 39.3%, 0.420)",
  limeA7: "hsla(79, 99.7%, 34.6%, 0.585)",
  limeA8: "hsla(76, 99.8%, 33.7%, 0.828)",
  limeA9: "hsla(81, 99.8%, 40.2%, 0.836)",
  limeA10: "hsla(80, 100%, 37.6%, 0.851)",
  limeA11: "hsla(75, 99.5%, 22.0%, 0.950)",
  limeA12: "hsla(78, 99.6%, 8.4%, 0.965)"
};
