// @tamagui/colors/dist/cjs/light/mauveA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var mauveA_exports = {};
__export(mauveA_exports, {
  mauveA: () => mauveA
});
module.exports = __toCommonJS(mauveA_exports);
const mauveA = {
  mauveA1: "hsla(300, 89.3%, 18.3%, 0.012)",
  mauveA2: "hsla(300, 78.1%, 9.0%, 0.028)",
  mauveA3: "hsla(300, 99.5%, 7.7%, 0.051)",
  mauveA4: "hsla(270, 90.5%, 6.1%, 0.071)",
  mauveA5: "hsla(270, 83.0%, 5.2%, 0.091)",
  mauveA6: "hsla(300, 93.5%, 3.7%, 0.114)",
  mauveA7: "hsla(270, 82.6%, 3.3%, 0.142)",
  mauveA8: "hsla(255, 95.2%, 3.7%, 0.220)",
  mauveA9: "hsla(255, 94.8%, 3.7%, 0.444)",
  mauveA10: "hsla(253, 96.5%, 3.8%, 0.483)",
  mauveA11: "hsla(247, 97.9%, 3.2%, 0.569)",
  mauveA12: "hsla(261, 98.7%, 3.0%, 0.918)"
};
