// @tamagui/colors/dist/cjs/light/oliveA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var oliveA_exports = {};
__export(oliveA_exports, {
  oliveA: () => oliveA
});
module.exports = __toCommonJS(oliveA_exports);
const oliveA = {
  oliveA1: "hsla(120, 89.3%, 18.3%, 0.012)",
  oliveA2: "hsla(120, 87.7%, 16.0%, 0.028)",
  oliveA3: "hsla(120, 99.5%, 7.7%, 0.051)",
  oliveA4: "hsla(120, 92.3%, 8.5%, 0.075)",
  oliveA5: "hsla(120, 86.0%, 6.9%, 0.099)",
  oliveA6: "hsla(120, 94.8%, 6.8%, 0.122)",
  oliveA7: "hsla(120, 99.3%, 5.2%, 0.153)",
  oliveA8: "hsla(110, 93.8%, 5.2%, 0.240)",
  oliveA9: "hsla(111, 98.7%, 3.0%, 0.459)",
  oliveA10: "hsla(111, 93.5%, 2.9%, 0.499)",
  oliveA11: "hsla(111, 95.2%, 2.5%, 0.585)",
  oliveA12: "hsla(110, 97.6%, 2.6%, 0.930)"
};
