// @tamagui/colors/dist/cjs/light/plumA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var plumA_exports = {};
__export(plumA_exports, {
  plumA: () => plumA
});
module.exports = __toCommonJS(plumA_exports);
const plumA = {
  plumA1: "hsla(280, 100%, 51.0%, 0.012)",
  plumA2: "hsla(300, 100%, 51.0%, 0.028)",
  plumA3: "hsla(300, 99.0%, 40.9%, 0.063)",
  plumA4: "hsla(300, 99.9%, 38.5%, 0.102)",
  plumA5: "hsla(298, 98.2%, 35.9%, 0.150)",
  plumA6: "hsla(297, 99.6%, 33.7%, 0.216)",
  plumA7: "hsla(295, 99.7%, 32.6%, 0.314)",
  plumA8: "hsla(292, 99.6%, 32.4%, 0.432)",
  plumA9: "hsla(292, 99.9%, 31.0%, 0.710)",
  plumA10: "hsla(292, 99.9%, 30.8%, 0.765)",
  plumA11: "hsla(292, 99.8%, 30.7%, 0.832)",
  plumA12: "hsla(291, 99.9%, 9.7%, 0.953)"
};
