// @tamagui/colors/dist/cjs/light/purpleA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var purpleA_exports = {};
__export(purpleA_exports, {
  purpleA: () => purpleA
});
module.exports = __toCommonJS(purpleA_exports);
const purpleA = {
  purpleA1: "hsla(300, 94.3%, 34.6%, 0.012)",
  purpleA2: "hsla(276, 100%, 51.0%, 0.020)",
  purpleA3: "hsla(277, 99.6%, 46.5%, 0.055)",
  purpleA4: "hsla(274, 97.9%, 44.3%, 0.095)",
  purpleA5: "hsla(276, 98.6%, 42.0%, 0.142)",
  purpleA6: "hsla(275, 100%, 39.2%, 0.200)",
  purpleA7: "hsla(273, 99.2%, 38.2%, 0.295)",
  purpleA8: "hsla(272, 99.7%, 37.6%, 0.424)",
  purpleA9: "hsla(272, 99.6%, 34.0%, 0.695)",
  purpleA10: "hsla(272, 99.7%, 32.0%, 0.730)",
  purpleA11: "hsla(272, 99.8%, 29.7%, 0.773)",
  purpleA12: "hsla(272, 99.2%, 11.3%, 0.946)"
};
