// @tamagui/colors/dist/cjs/light/redA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var redA_exports = {};
__export(redA_exports, {
  redA: () => redA
});
module.exports = __toCommonJS(redA_exports);
const redA = {
  redA1: "hsla(0, 100%, 51.0%, 0.012)",
  redA2: "hsla(0, 100%, 51.0%, 0.032)",
  redA3: "hsla(0, 100%, 50.2%, 0.063)",
  redA4: "hsla(0, 100%, 50.0%, 0.102)",
  redA5: "hsla(0, 99.9%, 47.5%, 0.153)",
  redA6: "hsla(0, 99.5%, 44.9%, 0.224)",
  redA7: "hsla(359, 99.7%, 42.7%, 0.318)",
  redA8: "hsla(359, 99.6%, 41.1%, 0.436)",
  redA9: "hsla(358, 99.9%, 42.9%, 0.718)",
  redA10: "hsla(358, 99.9%, 41.0%, 0.761)",
  redA11: "hsla(358, 99.8%, 38.3%, 0.832)",
  redA12: "hsla(355, 99.3%, 7.9%, 0.926)"
};
