// @tamagui/colors/dist/cjs/light/sage.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var sage_exports = {};
__export(sage_exports, {
  sage: () => sage
});
module.exports = __toCommonJS(sage_exports);
const sage = {
  sage1: "hsl(155, 30.0%, 98.8%)",
  sage2: "hsl(150, 16.7%, 97.6%)",
  sage3: "hsl(151, 10.6%, 95.2%)",
  sage4: "hsl(151, 8.8%, 93.0%)",
  sage5: "hsl(151, 7.8%, 90.8%)",
  sage6: "hsl(152, 7.2%, 88.4%)",
  sage7: "hsl(153, 6.7%, 85.3%)",
  sage8: "hsl(154, 6.1%, 77.5%)",
  sage9: "hsl(155, 3.5%, 55.5%)",
  sage10: "hsl(154, 2.8%, 51.7%)",
  sage11: "hsl(155, 3.0%, 43.0%)",
  sage12: "hsl(155, 24.0%, 9.0%)"
};
