// @tamagui/colors/dist/cjs/light/sageA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var sageA_exports = {};
__export(sageA_exports, {
  sageA: () => sageA
});
module.exports = __toCommonJS(sageA_exports);
const sageA = {
  sageA1: "hsla(150, 92.6%, 26.5%, 0.016)",
  sageA2: "hsla(150, 87.7%, 16.0%, 0.028)",
  sageA3: "hsla(160, 98.4%, 10.9%, 0.055)",
  sageA4: "hsla(140, 92.3%, 8.5%, 0.075)",
  sageA5: "hsla(160, 86.0%, 6.9%, 0.099)",
  sageA6: "hsla(156, 95.1%, 8.2%, 0.126)",
  sageA7: "hsla(156, 98.6%, 6.3%, 0.157)",
  sageA8: "hsla(154, 94.6%, 6.0%, 0.240)",
  sageA9: "hsla(154, 98.7%, 3.0%, 0.459)",
  sageA10: "hsla(154, 93.5%, 2.9%, 0.499)",
  sageA11: "hsla(154, 95.2%, 2.5%, 0.585)",
  sageA12: "hsla(158, 97.0%, 2.4%, 0.934)"
};
