// @tamagui/colors/dist/cjs/light/slateA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var slateA_exports = {};
__export(slateA_exports, {
  slateA: () => slateA
});
module.exports = __toCommonJS(slateA_exports);
const slateA = {
  slateA1: "hsla(210, 92.6%, 26.5%, 0.016)",
  slateA2: "hsla(210, 87.7%, 16.0%, 0.028)",
  slateA3: "hsla(210, 98.8%, 14.4%, 0.055)",
  slateA4: "hsla(210, 94.1%, 11.1%, 0.075)",
  slateA5: "hsla(216, 91.1%, 10.9%, 0.099)",
  slateA6: "hsla(206, 96.4%, 11.3%, 0.126)",
  slateA7: "hsla(210, 99.1%, 10.1%, 0.157)",
  slateA8: "hsla(205, 96.5%, 10.0%, 0.244)",
  slateA9: "hsla(206, 98.8%, 5.9%, 0.467)",
  slateA10: "hsla(206, 99.6%, 5.4%, 0.506)",
  slateA11: "hsla(206, 97.0%, 4.8%, 0.593)",
  slateA12: "hsla(202, 97.0%, 2.4%, 0.934)"
};
