// @tamagui/colors/dist/cjs/light/tealA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var tealA_exports = {};
__export(tealA_exports, {
  tealA: () => tealA
});
module.exports = __toCommonJS(tealA_exports);
const tealA = {
  tealA1: "hsla(165, 95.2%, 41.2%, 0.020)",
  tealA2: "hsla(169, 99.5%, 39.4%, 0.055)",
  tealA3: "hsla(167, 97.6%, 38.1%, 0.095)",
  tealA4: "hsla(168, 98.1%, 34.6%, 0.150)",
  tealA5: "hsla(170, 99.4%, 32.3%, 0.220)",
  tealA6: "hsla(170, 99.7%, 30.1%, 0.314)",
  tealA7: "hsla(170, 99.3%, 28.7%, 0.448)",
  tealA8: "hsla(172, 99.8%, 29.7%, 0.675)",
  tealA9: "hsla(173, 99.8%, 31.1%, 0.930)",
  tealA10: "hsla(173, 99.7%, 28.7%, 0.946)",
  tealA11: "hsla(174, 99.8%, 23.3%, 0.977)",
  tealA12: "hsla(171, 98.8%, 6.8%, 0.938)"
};
