// @tamagui/colors/dist/cjs/light/teal.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var teal_exports = {};
__export(teal_exports, {
  teal: () => teal
});
module.exports = __toCommonJS(teal_exports);
const teal = {
  teal1: "hsl(165, 60.0%, 98.8%)",
  teal2: "hsl(169, 64.7%, 96.7%)",
  teal3: "hsl(169, 59.8%, 94.0%)",
  teal4: "hsl(169, 53.1%, 90.2%)",
  teal5: "hsl(170, 47.1%, 85.0%)",
  teal6: "hsl(170, 42.6%, 77.9%)",
  teal7: "hsl(170, 39.9%, 68.1%)",
  teal8: "hsl(172, 42.1%, 52.5%)",
  teal9: "hsl(173, 80.0%, 36.0%)",
  teal10: "hsl(173, 83.4%, 32.5%)",
  teal11: "hsl(174, 90.0%, 25.2%)",
  teal12: "hsl(170, 50.0%, 12.5%)"
};
