// @tamagui/colors/dist/cjs/light/tomatoA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var tomatoA_exports = {};
__export(tomatoA_exports, {
  tomatoA: () => tomatoA
});
module.exports = __toCommonJS(tomatoA_exports);
const tomatoA = {
  tomatoA1: "hsla(0, 100%, 51.0%, 0.012)",
  tomatoA2: "hsla(8, 100%, 51.0%, 0.032)",
  tomatoA3: "hsla(7, 100%, 50.2%, 0.067)",
  tomatoA4: "hsla(8, 100%, 50.1%, 0.114)",
  tomatoA5: "hsla(7, 99.5%, 47.9%, 0.173)",
  tomatoA6: "hsla(9, 99.9%, 46.2%, 0.255)",
  tomatoA7: "hsla(10, 99.8%, 43.6%, 0.365)",
  tomatoA8: "hsla(10, 99.5%, 41.8%, 0.499)",
  tomatoA9: "hsla(10, 99.9%, 43.8%, 0.820)",
  tomatoA10: "hsla(10, 100%, 41.8%, 0.859)",
  tomatoA11: "hsla(10, 99.9%, 38.8%, 0.922)",
  tomatoA12: "hsla(10, 99.0%, 7.4%, 0.934)"
};
