// @tamagui/colors/dist/cjs/light/violetA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var violetA_exports = {};
__export(violetA_exports, {
  violetA: () => violetA
});
module.exports = __toCommonJS(violetA_exports);
const violetA = {
  violetA1: "hsla(270, 94.3%, 34.6%, 0.012)",
  violetA2: "hsla(252, 100%, 51.0%, 0.020)",
  violetA3: "hsla(254, 100%, 50.0%, 0.051)",
  violetA4: "hsla(251, 98.3%, 48.2%, 0.087)",
  violetA5: "hsla(252, 99.0%, 45.7%, 0.130)",
  violetA6: "hsla(251, 99.1%, 44.0%, 0.189)",
  violetA7: "hsla(252, 99.5%, 41.7%, 0.279)",
  violetA8: "hsla(252, 100%, 40.7%, 0.400)",
  violetA9: "hsla(252, 99.9%, 35.8%, 0.663)",
  violetA10: "hsla(251, 99.6%, 32.5%, 0.691)",
  violetA11: "hsla(250, 99.8%, 28.4%, 0.726)",
  violetA12: "hsla(254, 99.5%, 11.9%, 0.926)"
};
