// @tamagui/colors/dist/cjs/light/yellowA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var yellowA_exports = {};
__export(yellowA_exports, {
  yellowA: () => yellowA
});
module.exports = __toCommonJS(yellowA_exports);
const yellowA = {
  yellowA1: "hsla(60, 94.3%, 34.6%, 0.024)",
  yellowA2: "hsla(52, 100%, 50.4%, 0.091)",
  yellowA3: "hsla(55, 100%, 50.2%, 0.181)",
  yellowA4: "hsla(54, 100%, 50.1%, 0.267)",
  yellowA5: "hsla(52, 99.9%, 49.5%, 0.357)",
  yellowA6: "hsla(50, 100%, 47.4%, 0.451)",
  yellowA7: "hsla(47, 99.8%, 44.6%, 0.577)",
  yellowA8: "hsla(48, 100%, 46.0%, 0.980)",
  yellowA9: "hsla(53, 100%, 48.0%, 0.961)",
  yellowA10: "hsla(50, 100%, 48.4%, 0.980)",
  yellowA11: "hsla(42, 100%, 28.6%, 0.980)",
  yellowA12: "hsla(41, 98.9%, 8.0%, 0.942)"
};
