// @tamagui/web/dist/cjs/helpers/createEmptyAnimationDriver.native.js
"use strict";

var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var createEmptyAnimationDriver_exports = {};
__export(createEmptyAnimationDriver_exports, {
  createEmptyAnimationDriver: () => createEmptyAnimationDriver
});
module.exports = __toCommonJS(createEmptyAnimationDriver_exports);
var noAnimationDriver = function (method) {
    throw new Error(`No animation driver configured. To use ${method}, you must pass \`animations\` to createTamagui. See: https://tamagui.dev/docs/core/animations`);
  },
  createEmptyAnimationDriver = function () {
    return {
      isReactNative: !1,
      animations: {},
      useAnimations: function () {
        return noAnimationDriver("animations");
      },
      usePresence: function () {
        return noAnimationDriver("usePresence");
      },
      ResetPresence: function () {
        return noAnimationDriver("ResetPresence");
      },
      useAnimatedNumber: function () {
        return noAnimationDriver("useAnimatedNumber");
      },
      useAnimatedNumberStyle: function () {
        return noAnimationDriver("useAnimatedNumberStyle");
      },
      useAnimatedNumberReaction: function () {
        return noAnimationDriver("useAnimatedNumberReaction");
      }
    };
  };
