// @tamagui/web/dist/cjs/helpers/webPropsToSkip.native.js
"use strict";

var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var webPropsToSkip_native_exports = {};
__export(webPropsToSkip_native_exports, {
  webPropsToSkip: () => webPropsToSkip
});
module.exports = __toCommonJS(webPropsToSkip_native_exports);
var webPropsToSkip = {
  whiteSpace: 1,
  wordWrap: 1,
  textOverflow: 1,
  textDecorationDistance: 1,
  cursor: 1,
  contain: 1,
  boxSizing: 1,
  touchAction: 1,
  outlineStyle: 1,
  outlineOffset: 1,
  outlineWidth: 1,
  outlineColor: 1,
  backdropFilter: 1,
  backgroundImage: 1,
  mixBlendMode: 1,
  scrollbarWidth: 1,
  backgroundOrigin: 1,
  backgroundPosition: 1,
  backgroundRepeat: 1,
  backgroundSize: 1,
  backgroundClip: 1,
  backgroundBlendMode: 1,
  backgroundAttachment: 1,
  background: 1,
  clipPath: 1,
  caretColor: 1,
  transformStyle: 1,
  mask: 1,
  maskImage: 1,
  textEmphasis: 1,
  borderImage: 1,
  float: 1,
  content: 1,
  overflowBlock: 1,
  overflowInline: 1,
  maskBorder: 1,
  maskBorderMode: 1,
  maskBorderOutset: 1,
  maskBorderRepeat: 1,
  maskBorderSlice: 1,
  maskBorderSource: 1,
  maskBorderWidth: 1,
  maskClip: 1,
  maskComposite: 1,
  maskMode: 1,
  maskOrigin: 1,
  maskPosition: 1,
  maskRepeat: 1,
  maskSize: 1,
  maskType: 1
};
