// @tamagui/web/dist/cjs/index.native.js
"use strict";

var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  },
  __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var index_exports = {};
__export(index_exports, {
  _disableMediaTouch: () => import_useMedia._disableMediaTouch,
  configureMedia: () => import_useMedia.configureMedia,
  forceUpdateThemes: () => import_useThemeState.forceUpdateThemes,
  getConfig: () => import_config.getConfig,
  getMedia: () => import_useMedia.getMedia,
  getSetting: () => import_config.getSetting,
  getThemes: () => import_config.getThemes,
  getToken: () => import_config.getToken,
  getTokenValue: () => import_config.getTokenValue,
  getTokens: () => import_config.getTokens,
  insertStyleRules: () => import_insertStyleRule.insertStyleRules,
  mediaKeyMatch: () => import_useMedia.mediaKeyMatch,
  mediaObjectToString: () => import_useMedia.mediaObjectToString,
  mediaQueryConfig: () => import_useMedia.mediaQueryConfig,
  mediaState: () => import_useMedia.mediaState,
  setConfig: () => import_config.setConfig,
  setNonce: () => import_insertStyleRule2.setNonce,
  setupDev: () => import_config.setupDev,
  updateConfig: () => import_config.updateConfig,
  useMedia: () => import_useMedia.useMedia
});
module.exports = __toCommonJS(index_exports);
__reExport(index_exports, require("./contexts/ComponentContext.native.js"), module.exports);
__reExport(index_exports, require("./contexts/GroupContext.native.js"), module.exports);
__reExport(index_exports, require("@tamagui/is-equal-shallow"), module.exports);
__reExport(index_exports, require("./_withStableStyle.native.js"), module.exports);
__reExport(index_exports, require("./createComponent.native.js"), module.exports);
__reExport(index_exports, require("./helpers/createMediaStyle.native.js"), module.exports);
__reExport(index_exports, require("./helpers/createStyledContext.native.js"), module.exports);
__reExport(index_exports, require("./helpers/expandStyles.native.js"), module.exports);
__reExport(index_exports, require("./helpers/getCSSStylesAtomic.native.js"), module.exports);
__reExport(index_exports, require("./helpers/getExpandedShorthands.native.js"), module.exports);
__reExport(index_exports, require("./helpers/getShorthandValue.native.js"), module.exports);
__reExport(index_exports, require("./helpers/getSplitStyles.native.js"), module.exports);
__reExport(index_exports, require("./helpers/getThemeCSSRules.native.js"), module.exports);
__reExport(index_exports, require("./helpers/getVariantExtras.native.js"), module.exports);
var import_insertStyleRule = require("./helpers/insertStyleRule.native.js");
__reExport(index_exports, require("./helpers/isTamaguiComponent.native.js"), module.exports);
__reExport(index_exports, require("./helpers/isTamaguiElement.native.js"), module.exports);
__reExport(index_exports, require("./helpers/matchMedia.native.js"), module.exports);
__reExport(index_exports, require("./helpers/mergeProps.native.js"), module.exports);
__reExport(index_exports, require("./helpers/normalizeColor.native.js"), module.exports);
__reExport(index_exports, require("./helpers/normalizeStyle.native.js"), module.exports);
__reExport(index_exports, require("./helpers/normalizeValueWithProperty.native.js"), module.exports);
__reExport(index_exports, require("./helpers/propMapper.native.js"), module.exports);
__reExport(index_exports, require("./helpers/proxyThemeToParents.native.js"), module.exports);
__reExport(index_exports, require("./helpers/proxyThemeVariables.native.js"), module.exports);
__reExport(index_exports, require("./helpers/pseudoDescriptors.native.js"), module.exports);
__reExport(index_exports, require("./helpers/themeable.native.js"), module.exports);
__reExport(index_exports, require("./helpers/themes.native.js"), module.exports);
__reExport(index_exports, require("./helpers/transformsToString.native.js"), module.exports);
__reExport(index_exports, require("./helpers/wrapStyleTags.native.js"), module.exports);
__reExport(index_exports, require("./createComponent.native.js"), module.exports);
__reExport(index_exports, require("./createFont.native.js"), module.exports);
__reExport(index_exports, require("./createShorthands.native.js"), module.exports);
__reExport(index_exports, require("./createTamagui.native.js"), module.exports);
__reExport(index_exports, require("./createTheme.native.js"), module.exports);
__reExport(index_exports, require("./createTokens.native.js"), module.exports);
__reExport(index_exports, require("./createVariable.native.js"), module.exports);
__reExport(index_exports, require("./createVariables.native.js"), module.exports);
__reExport(index_exports, require("./insertFont.native.js"), module.exports);
__reExport(index_exports, require("./setupReactNative.native.js"), module.exports);
__reExport(index_exports, require("./styled.native.js"), module.exports);
__reExport(index_exports, require("./Tamagui.native.js"), module.exports);
__reExport(index_exports, require("./interfaces/GetRef.native.js"), module.exports);
var import_config = require("./config.native.js"),
  import_insertStyleRule2 = require("./helpers/insertStyleRule.native.js");
__reExport(index_exports, require("./constants/constants.native.js"), module.exports);
__reExport(index_exports, require("./hooks/useIsTouchDevice.native.js"), module.exports);
var import_useMedia = require("./hooks/useMedia.native.js");
__reExport(index_exports, require("./hooks/useProps.native.js"), module.exports);
__reExport(index_exports, require("./hooks/useTheme.native.js"), module.exports);
__reExport(index_exports, require("./hooks/useThemeName.native.js"), module.exports);
var import_useThemeState = require("./hooks/useThemeState.native.js");
__reExport(index_exports, require("./views/Configuration.native.js"), module.exports);
__reExport(index_exports, require("./views/FontLanguage.native.js"), module.exports);
__reExport(index_exports, require("./views/Slot.native.js"), module.exports);
__reExport(index_exports, require("./views/Stack.native.js"), module.exports);
__reExport(index_exports, require("./views/TamaguiProvider.native.js"), module.exports);
__reExport(index_exports, require("./views/Text.native.js"), module.exports);
__reExport(index_exports, require("./views/Theme.native.js"), module.exports);
__reExport(index_exports, require("./views/ThemeProvider.native.js"), module.exports);
__reExport(index_exports, require("./views/View.native.js"), module.exports);
__reExport(index_exports, require("@tamagui/compose-refs"), module.exports);
__reExport(index_exports, require("@tamagui/constants"), module.exports);
__reExport(index_exports, require("@tamagui/helpers"), module.exports);
__reExport(index_exports, require("@tamagui/use-did-finish-ssr"), module.exports);
__reExport(index_exports, require("@tamagui/use-event"), module.exports);
__reExport(index_exports, require("./setupHooks.native.js"), module.exports);
