// @tamagui/web/dist/cjs/index.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  },
  __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var index_exports = {};
__export(index_exports, {
  _disableMediaTouch: () => import_useMedia._disableMediaTouch,
  configureMedia: () => import_useMedia.configureMedia,
  forceUpdateThemes: () => import_useThemeState.forceUpdateThemes,
  getConfig: () => import_config.getConfig,
  getMedia: () => import_useMedia.getMedia,
  getSetting: () => import_config.getSetting,
  getThemes: () => import_config.getThemes,
  getToken: () => import_config.getToken,
  getTokenValue: () => import_config.getTokenValue,
  getTokens: () => import_config.getTokens,
  insertStyleRules: () => import_insertStyleRule.insertStyleRules,
  mediaKeyMatch: () => import_useMedia.mediaKeyMatch,
  mediaObjectToString: () => import_useMedia.mediaObjectToString,
  mediaQueryConfig: () => import_useMedia.mediaQueryConfig,
  mediaState: () => import_useMedia.mediaState,
  setConfig: () => import_config.setConfig,
  setNonce: () => import_insertStyleRule2.setNonce,
  setupDev: () => import_config.setupDev,
  updateConfig: () => import_config.updateConfig,
  useMedia: () => import_useMedia.useMedia
});
module.exports = __toCommonJS(index_exports);
__reExport(index_exports, require("./contexts/ComponentContext.cjs"), module.exports);
__reExport(index_exports, require("./contexts/GroupContext.cjs"), module.exports);
__reExport(index_exports, require("@tamagui/is-equal-shallow"), module.exports);
__reExport(index_exports, require("./_withStableStyle.cjs"), module.exports);
__reExport(index_exports, require("./createComponent.cjs"), module.exports);
__reExport(index_exports, require("./helpers/createMediaStyle.cjs"), module.exports);
__reExport(index_exports, require("./helpers/createStyledContext.cjs"), module.exports);
__reExport(index_exports, require("./helpers/expandStyles.cjs"), module.exports);
__reExport(index_exports, require("./helpers/getCSSStylesAtomic.cjs"), module.exports);
__reExport(index_exports, require("./helpers/getExpandedShorthands.cjs"), module.exports);
__reExport(index_exports, require("./helpers/getShorthandValue.cjs"), module.exports);
__reExport(index_exports, require("./helpers/getSplitStyles.cjs"), module.exports);
__reExport(index_exports, require("./helpers/getThemeCSSRules.cjs"), module.exports);
__reExport(index_exports, require("./helpers/getVariantExtras.cjs"), module.exports);
var import_insertStyleRule = require("./helpers/insertStyleRule.cjs");
__reExport(index_exports, require("./helpers/isTamaguiComponent.cjs"), module.exports);
__reExport(index_exports, require("./helpers/isTamaguiElement.cjs"), module.exports);
__reExport(index_exports, require("./helpers/matchMedia.cjs"), module.exports);
__reExport(index_exports, require("./helpers/mergeProps.cjs"), module.exports);
__reExport(index_exports, require("./helpers/normalizeColor.cjs"), module.exports);
__reExport(index_exports, require("./helpers/normalizeStyle.cjs"), module.exports);
__reExport(index_exports, require("./helpers/normalizeValueWithProperty.cjs"), module.exports);
__reExport(index_exports, require("./helpers/propMapper.cjs"), module.exports);
__reExport(index_exports, require("./helpers/proxyThemeToParents.cjs"), module.exports);
__reExport(index_exports, require("./helpers/proxyThemeVariables.cjs"), module.exports);
__reExport(index_exports, require("./helpers/pseudoDescriptors.cjs"), module.exports);
__reExport(index_exports, require("./helpers/themeable.cjs"), module.exports);
__reExport(index_exports, require("./helpers/themes.cjs"), module.exports);
__reExport(index_exports, require("./helpers/transformsToString.cjs"), module.exports);
__reExport(index_exports, require("./helpers/wrapStyleTags.cjs"), module.exports);
__reExport(index_exports, require("./createComponent.cjs"), module.exports);
__reExport(index_exports, require("./createFont.cjs"), module.exports);
__reExport(index_exports, require("./createShorthands.cjs"), module.exports);
__reExport(index_exports, require("./createTamagui.cjs"), module.exports);
__reExport(index_exports, require("./createTheme.cjs"), module.exports);
__reExport(index_exports, require("./createTokens.cjs"), module.exports);
__reExport(index_exports, require("./createVariable.cjs"), module.exports);
__reExport(index_exports, require("./createVariables.cjs"), module.exports);
__reExport(index_exports, require("./insertFont.cjs"), module.exports);
__reExport(index_exports, require("./setupReactNative.cjs"), module.exports);
__reExport(index_exports, require("./styled.cjs"), module.exports);
__reExport(index_exports, require("./Tamagui.cjs"), module.exports);
__reExport(index_exports, require("./interfaces/GetRef.cjs"), module.exports);
var import_config = require("./config.cjs"),
  import_insertStyleRule2 = require("./helpers/insertStyleRule.cjs");
__reExport(index_exports, require("./constants/constants.cjs"), module.exports);
__reExport(index_exports, require("./hooks/useIsTouchDevice.cjs"), module.exports);
var import_useMedia = require("./hooks/useMedia.cjs");
__reExport(index_exports, require("./hooks/useProps.cjs"), module.exports);
__reExport(index_exports, require("./hooks/useTheme.cjs"), module.exports);
__reExport(index_exports, require("./hooks/useThemeName.cjs"), module.exports);
var import_useThemeState = require("./hooks/useThemeState.cjs");
__reExport(index_exports, require("./views/Configuration.cjs"), module.exports);
__reExport(index_exports, require("./views/FontLanguage.cjs"), module.exports);
__reExport(index_exports, require("./views/Slot.cjs"), module.exports);
__reExport(index_exports, require("./views/Stack.cjs"), module.exports);
__reExport(index_exports, require("./views/TamaguiProvider.cjs"), module.exports);
__reExport(index_exports, require("./views/Text.cjs"), module.exports);
__reExport(index_exports, require("./views/Theme.cjs"), module.exports);
__reExport(index_exports, require("./views/ThemeProvider.cjs"), module.exports);
__reExport(index_exports, require("./views/View.cjs"), module.exports);
__reExport(index_exports, require("@tamagui/compose-refs"), module.exports);
__reExport(index_exports, require("@tamagui/constants"), module.exports);
__reExport(index_exports, require("@tamagui/helpers"), module.exports);
__reExport(index_exports, require("@tamagui/use-did-finish-ssr"), module.exports);
__reExport(index_exports, require("@tamagui/use-event"), module.exports);
__reExport(index_exports, require("./setupHooks.cjs"), module.exports);
