// @tamagui/colors/dist/cjs/index.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var index_exports = {};
__export(index_exports, {
  amber: () => import_amber2.amber,
  amberA: () => import_amberA2.amberA,
  amberDark: () => import_amber.amber,
  amberDarkA: () => import_amberA.amberA,
  blackA: () => import_blackA.blackA,
  blue: () => import_blue2.blue,
  blueA: () => import_blueA2.blueA,
  blueDark: () => import_blue.blue,
  blueDarkA: () => import_blueA.blueA,
  bronze: () => import_bronze2.bronze,
  bronzeA: () => import_bronzeA2.bronzeA,
  bronzeDark: () => import_bronze.bronze,
  bronzeDarkA: () => import_bronzeA.bronzeA,
  brown: () => import_brown2.brown,
  brownA: () => import_brownA2.brownA,
  brownDark: () => import_brown.brown,
  brownDarkA: () => import_brownA.brownA,
  crimson: () => import_crimson2.crimson,
  crimsonA: () => import_crimsonA2.crimsonA,
  crimsonDark: () => import_crimson.crimson,
  crimsonDarkA: () => import_crimsonA.crimsonA,
  cyan: () => import_cyan2.cyan,
  cyanA: () => import_cyanA2.cyanA,
  cyanDark: () => import_cyan.cyan,
  cyanDarkA: () => import_cyanA.cyanA,
  gold: () => import_gold2.gold,
  goldA: () => import_goldA2.goldA,
  goldDark: () => import_gold.gold,
  goldDarkA: () => import_goldA.goldA,
  grass: () => import_grass2.grass,
  grassA: () => import_grassA2.grassA,
  grassDark: () => import_grass.grass,
  grassDarkA: () => import_grassA.grassA,
  gray: () => import_gray2.gray,
  grayA: () => import_grayA2.grayA,
  grayDark: () => import_gray.gray,
  grayDarkA: () => import_grayA.grayA,
  green: () => import_green2.green,
  greenA: () => import_greenA2.greenA,
  greenDark: () => import_green.green,
  greenDarkA: () => import_greenA.greenA,
  indigo: () => import_indigo2.indigo,
  indigoA: () => import_indigoA2.indigoA,
  indigoDark: () => import_indigo.indigo,
  indigoDarkA: () => import_indigoA.indigoA,
  lime: () => import_lime2.lime,
  limeA: () => import_limeA2.limeA,
  limeDark: () => import_lime.lime,
  limeDarkA: () => import_limeA.limeA,
  mauve: () => import_mauve2.mauve,
  mauveA: () => import_mauveA2.mauveA,
  mauveDark: () => import_mauve.mauve,
  mauveDarkA: () => import_mauveA.mauveA,
  mint: () => import_mint2.mint,
  mintA: () => import_mintA2.mintA,
  mintDark: () => import_mint.mint,
  mintDarkA: () => import_mintA.mintA,
  olive: () => import_olive2.olive,
  oliveA: () => import_oliveA2.oliveA,
  oliveDark: () => import_olive.olive,
  oliveDarkA: () => import_oliveA.oliveA,
  orange: () => import_orange2.orange,
  orangeA: () => import_orangeA2.orangeA,
  orangeDark: () => import_orange.orange,
  orangeDarkA: () => import_orangeA.orangeA,
  pink: () => import_pink2.pink,
  pinkA: () => import_pinkA2.pinkA,
  pinkDark: () => import_pink.pink,
  pinkDarkA: () => import_pinkA.pinkA,
  plum: () => import_plum2.plum,
  plumA: () => import_plumA2.plumA,
  plumDark: () => import_plum.plum,
  plumDarkA: () => import_plumA.plumA,
  purple: () => import_purple2.purple,
  purpleA: () => import_purpleA2.purpleA,
  purpleDark: () => import_purple.purple,
  purpleDarkA: () => import_purpleA.purpleA,
  red: () => import_red2.red,
  redA: () => import_redA2.redA,
  redDark: () => import_red.red,
  redDarkA: () => import_redA.redA,
  sage: () => import_sage2.sage,
  sageA: () => import_sageA2.sageA,
  sageDark: () => import_sage.sage,
  sageDarkA: () => import_sageA.sageA,
  sand: () => import_sand2.sand,
  sandA: () => import_sandA2.sandA,
  sandDark: () => import_sand.sand,
  sandDarkA: () => import_sandA.sandA,
  sky: () => import_sky2.sky,
  skyA: () => import_skyA2.skyA,
  skyDark: () => import_sky.sky,
  skyDarkA: () => import_skyA.skyA,
  slate: () => import_slate2.slate,
  slateA: () => import_slateA2.slateA,
  slateDark: () => import_slate.slate,
  slateDarkA: () => import_slateA.slateA,
  teal: () => import_teal2.teal,
  tealA: () => import_tealA2.tealA,
  tealDark: () => import_teal.teal,
  tealDarkA: () => import_tealA.tealA,
  tomato: () => import_tomato2.tomato,
  tomatoA: () => import_tomatoA2.tomatoA,
  tomatoDark: () => import_tomato.tomato,
  tomatoDarkA: () => import_tomatoA.tomatoA,
  violet: () => import_violet2.violet,
  violetA: () => import_violetA2.violetA,
  violetDark: () => import_violet.violet,
  violetDarkA: () => import_violetA.violetA,
  whiteA: () => import_whiteA.whiteA,
  yellow: () => import_yellow2.yellow,
  yellowA: () => import_yellowA2.yellowA,
  yellowDark: () => import_yellow.yellow,
  yellowDarkA: () => import_yellowA.yellowA
});
module.exports = __toCommonJS(index_exports);
var import_amber = require("./dark/amber"), import_blue = require("./dark/blue"), import_bronze = require("./dark/bronze"), import_brown = require("./dark/brown"), import_crimson = require("./dark/crimson"), import_cyan = require("./dark/cyan"), import_gold = require("./dark/gold"), import_grass = require("./dark/grass"), import_gray = require("./dark/gray"), import_green = require("./dark/green"), import_indigo = require("./dark/indigo"), import_lime = require("./dark/lime"), import_mauve = require("./dark/mauve"), import_mint = require("./dark/mint"), import_olive = require("./dark/olive"), import_orange = require("./dark/orange"), import_pink = require("./dark/pink"), import_plum = require("./dark/plum"), import_purple = require("./dark/purple"), import_red = require("./dark/red"), import_sage = require("./dark/sage"), import_sand = require("./dark/sand"), import_sky = require("./dark/sky"), import_slate = require("./dark/slate"), import_teal = require("./dark/teal"), import_tomato = require("./dark/tomato"), import_violet = require("./dark/violet"), import_yellow = require("./dark/yellow"), import_amberA = require("./dark/amberA"), import_blueA = require("./dark/blueA"), import_bronzeA = require("./dark/bronzeA"), import_brownA = require("./dark/brownA"), import_crimsonA = require("./dark/crimsonA"), import_cyanA = require("./dark/cyanA"), import_goldA = require("./dark/goldA"), import_grassA = require("./dark/grassA"), import_grayA = require("./dark/grayA"), import_greenA = require("./dark/greenA"), import_indigoA = require("./dark/indigoA"), import_limeA = require("./dark/limeA"), import_mauveA = require("./dark/mauveA"), import_mintA = require("./dark/mintA"), import_oliveA = require("./dark/oliveA"), import_orangeA = require("./dark/orangeA"), import_pinkA = require("./dark/pinkA"), import_plumA = require("./dark/plumA"), import_purpleA = require("./dark/purpleA"), import_redA = require("./dark/redA"), import_sageA = require("./dark/sageA"), import_sandA = require("./dark/sandA"), import_skyA = require("./dark/skyA"), import_slateA = require("./dark/slateA"), import_tealA = require("./dark/tealA"), import_tomatoA = require("./dark/tomatoA"), import_violetA = require("./dark/violetA"), import_yellowA = require("./dark/yellowA"), import_amber2 = require("./light/amber"), import_blue2 = require("./light/blue"), import_bronze2 = require("./light/bronze"), import_brown2 = require("./light/brown"), import_crimson2 = require("./light/crimson"), import_cyan2 = require("./light/cyan"), import_gold2 = require("./light/gold"), import_grass2 = require("./light/grass"), import_gray2 = require("./light/gray"), import_green2 = require("./light/green"), import_indigo2 = require("./light/indigo"), import_lime2 = require("./light/lime"), import_mauve2 = require("./light/mauve"), import_mint2 = require("./light/mint"), import_olive2 = require("./light/olive"), import_orange2 = require("./light/orange"), import_pink2 = require("./light/pink"), import_plum2 = require("./light/plum"), import_purple2 = require("./light/purple"), import_red2 = require("./light/red"), import_sage2 = require("./light/sage"), import_sand2 = require("./light/sand"), import_sky2 = require("./light/sky"), import_slate2 = require("./light/slate"), import_teal2 = require("./light/teal"), import_tomato2 = require("./light/tomato"), import_violet2 = require("./light/violet"), import_yellow2 = require("./light/yellow"), import_amberA2 = require("./light/amberA"), import_blueA2 = require("./light/blueA"), import_bronzeA2 = require("./light/bronzeA"), import_brownA2 = require("./light/brownA"), import_crimsonA2 = require("./light/crimsonA"), import_cyanA2 = require("./light/cyanA"), import_goldA2 = require("./light/goldA"), import_grassA2 = require("./light/grassA"), import_grayA2 = require("./light/grayA"), import_greenA2 = require("./light/greenA"), import_indigoA2 = require("./light/indigoA"), import_limeA2 = require("./light/limeA"), import_mauveA2 = require("./light/mauveA"), import_mintA2 = require("./light/mintA"), import_oliveA2 = require("./light/oliveA"), import_orangeA2 = require("./light/orangeA"), import_pinkA2 = require("./light/pinkA"), import_plumA2 = require("./light/plumA"), import_purpleA2 = require("./light/purpleA"), import_redA2 = require("./light/redA"), import_sageA2 = require("./light/sageA"), import_sandA2 = require("./light/sandA"), import_skyA2 = require("./light/skyA"), import_slateA2 = require("./light/slateA"), import_tealA2 = require("./light/tealA"), import_tomatoA2 = require("./light/tomatoA"), import_violetA2 = require("./light/violetA"), import_yellowA2 = require("./light/yellowA"), import_blackA = require("./blackA"), import_whiteA = require("./whiteA");
