// @tamagui/use-window-dimensions/dist/cjs/initialValue.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var initialValue_exports = {};
__export(initialValue_exports, {
  configureInitialWindowDimensions: () => configureInitialWindowDimensions,
  initialValue: () => initialValue
});
module.exports = __toCommonJS(initialValue_exports);
const initialValue = {
  width: 800,
  height: 600,
  scale: 1,
  fontScale: 1
};
function configureInitialWindowDimensions(next) {
  Object.assign(initialValue, next);
}
