// @tamagui/config/dist/cjs/animations.css.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var animations_css_exports = {};
__export(animations_css_exports, {
  animations: () => animations
});
module.exports = __toCommonJS(animations_css_exports);
var import_animations_css = require("@tamagui/animations-css");
const animations = (0, import_animations_css.createAnimations)({
  "100ms": "ease-in 100ms",
  bouncy: "ease-in 200ms",
  lazy: "ease-in 600ms",
  slow: "ease-in 500ms",
  medium: "ease-in 300ms",
  quick: "ease-in 100ms",
  tooltip: "ease-in 400ms"
});
