// @tamagui/config/dist/cjs/animationsReanimated.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var animationsReanimated_exports = {};
__export(animationsReanimated_exports, {
  animationsReanimated: () => animationsReanimated
});
module.exports = __toCommonJS(animationsReanimated_exports);
var import_animations_moti = require("@tamagui/animations-moti");
const animationsReanimated = (0, import_animations_moti.createAnimations)({
  "75ms": {
    type: "timing",
    duration: 75
  },
  "100ms": {
    type: "timing",
    duration: 100
  },
  "200ms": {
    type: "timing",
    duration: 200
  },
  superBouncy: {
    damping: 5,
    mass: 0.7,
    stiffness: 200
  },
  bouncy: {
    damping: 9,
    mass: 0.9,
    stiffness: 150
  },
  medium: {
    damping: 15,
    stiffness: 120,
    mass: 1
  },
  lazy: {
    damping: 18,
    stiffness: 50
  },
  slow: {
    damping: 15,
    stiffness: 40
  },
  quick: {
    damping: 20,
    mass: 1.2,
    stiffness: 250
  },
  tooltip: {
    damping: 10,
    mass: 0.9,
    stiffness: 100
  },
  quicker: {
    damping: 20,
    mass: 0.7,
    stiffness: 250
  },
  quickest: {
    damping: 5,
    mass: 0.2,
    stiffness: 300
  }
});
