// @tamagui/config/dist/cjs/animations.reanimated.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var animations_reanimated_exports = {};
__export(animations_reanimated_exports, {
  animations: () => animations
});
module.exports = __toCommonJS(animations_reanimated_exports);
var import_animations_moti = require("@tamagui/animations-moti");
const animations = (0, import_animations_moti.createAnimations)({
  "100ms": {
    type: "timing",
    duration: 100
  },
  bouncy: {
    type: "spring",
    damping: 9,
    mass: 0.9,
    stiffness: 150
  },
  lazy: {
    type: "spring",
    damping: 18,
    stiffness: 50
  },
  medium: {
    damping: 15,
    stiffness: 120,
    mass: 1
  },
  slow: {
    type: "spring",
    damping: 15,
    stiffness: 40
  },
  quick: {
    type: "spring",
    damping: 20,
    mass: 1.2,
    stiffness: 250
  },
  tooltip: {
    type: "spring",
    damping: 10,
    mass: 0.9,
    stiffness: 100
  }
});
