// @tamagui/config/dist/cjs/media.native.js
"use strict";

var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var media_exports = {};
__export(media_exports, {
  demoMedia: () => demoMedia,
  media: () => media,
  mediaQueryDefaultActive: () => mediaQueryDefaultActive,
  widths: () => widths
});
module.exports = __toCommonJS(media_exports);
var demoMedia = [500, 620, 780, 900],
  widths = [660, 800, 1020, 1280],
  media = {
    // for site
    xl: {
      maxWidth: 1650
    },
    lg: {
      maxWidth: 1280
    },
    md: {
      maxWidth: 1020
    },
    sm: {
      maxWidth: 800
    },
    xs: {
      maxWidth: 660
    },
    xxs: {
      maxWidth: 390
    },
    gtXs: {
      minWidth: 661
    },
    gtSm: {
      minWidth: 801
    },
    gtMd: {
      minWidth: 1021
    },
    gtLg: {
      minWidth: 1281
    },
    gtXl: {
      minWidth: 1651
    }
  },
  mediaQueryDefaultActive = {
    xl: !0,
    lg: !0,
    md: !0,
    sm: !0,
    xs: !0,
    // false
    xxs: !1
  };
