// @tamagui/config/dist/cjs/v2-base.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var v2_base_exports = {};
__export(v2_base_exports, {
  config: () => config
});
module.exports = __toCommonJS(v2_base_exports);
var import_shorthands = require("@tamagui/shorthands"), import_v2 = require("@tamagui/themes/v2"), import_v2_themes = require("@tamagui/themes/v2-themes"), import_fonts = require("./fonts"), import_media = require("./media");
const config = {
  themes: import_v2_themes.themes,
  media: import_media.media,
  shorthands: import_shorthands.shorthands,
  tokens: import_v2.tokens,
  fonts: import_fonts.fonts,
  selectionStyles: (theme) => theme.color5 ? {
    backgroundColor: theme.color5,
    color: theme.color11
  } : null,
  settings: {
    defaultFont: "body",
    shouldAddPrefersColorThemes: !0,
    themeClassNameOnRoot: !0,
    mediaQueryDefaultActive: import_media.mediaQueryDefaultActive
  }
};
