// @tamagui/config/dist/cjs/v4-media.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var v4_media_exports = {};
__export(v4_media_exports, {
  breakpoints: () => breakpoints,
  media: () => media,
  mediaQueryDefaultActive: () => mediaQueryDefaultActive
});
module.exports = __toCommonJS(v4_media_exports);
const breakpoints = {
  "2xl": 1536,
  xl: 1280,
  lg: 1024,
  md: 768,
  sm: 640,
  xs: 460,
  "2xs": 340
}, media = {
  maxXs: { maxWidth: breakpoints.xs },
  max2xs: { maxWidth: breakpoints["2xs"] },
  maxSm: { maxWidth: breakpoints.sm },
  maxMd: { maxWidth: breakpoints.md },
  maxLg: { maxWidth: breakpoints.lg },
  maxXl: { maxWidth: breakpoints.xl },
  max2Xl: { maxWidth: breakpoints["2xl"] },
  // for site
  "2xl": { minWidth: breakpoints["2xl"] },
  xl: { minWidth: breakpoints.xl },
  lg: { minWidth: breakpoints.lg },
  md: { minWidth: breakpoints.md },
  sm: { minWidth: breakpoints.sm },
  xs: { minWidth: breakpoints.xs },
  "2xs": { minWidth: breakpoints["2xs"] }
}, mediaQueryDefaultActive = {
  "2xl": !1,
  xl: !1,
  lg: !1,
  md: !1,
  sm: !1,
  xs: !0,
  "2xs": !0
};
