// @tamagui/colors/dist/esm/index.native.js
import { amber } from "./dark/amber.native.js";
import { blue } from "./dark/blue.native.js";
import { bronze } from "./dark/bronze.native.js";
import { brown } from "./dark/brown.native.js";
import { crimson } from "./dark/crimson.native.js";
import { cyan } from "./dark/cyan.native.js";
import { gold } from "./dark/gold.native.js";
import { grass } from "./dark/grass.native.js";
import { gray } from "./dark/gray.native.js";
import { green } from "./dark/green.native.js";
import { indigo } from "./dark/indigo.native.js";
import { lime } from "./dark/lime.native.js";
import { mauve } from "./dark/mauve.native.js";
import { mint } from "./dark/mint.native.js";
import { olive } from "./dark/olive.native.js";
import { orange } from "./dark/orange.native.js";
import { pink } from "./dark/pink.native.js";
import { plum } from "./dark/plum.native.js";
import { purple } from "./dark/purple.native.js";
import { red } from "./dark/red.native.js";
import { sage } from "./dark/sage.native.js";
import { sand } from "./dark/sand.native.js";
import { sky } from "./dark/sky.native.js";
import { slate } from "./dark/slate.native.js";
import { teal } from "./dark/teal.native.js";
import { tomato } from "./dark/tomato.native.js";
import { violet } from "./dark/violet.native.js";
import { yellow } from "./dark/yellow.native.js";
import { amberA } from "./dark/amberA.native.js";
import { blueA } from "./dark/blueA.native.js";
import { bronzeA } from "./dark/bronzeA.native.js";
import { brownA } from "./dark/brownA.native.js";
import { crimsonA } from "./dark/crimsonA.native.js";
import { cyanA } from "./dark/cyanA.native.js";
import { goldA } from "./dark/goldA.native.js";
import { grassA } from "./dark/grassA.native.js";
import { grayA } from "./dark/grayA.native.js";
import { greenA } from "./dark/greenA.native.js";
import { indigoA } from "./dark/indigoA.native.js";
import { limeA } from "./dark/limeA.native.js";
import { mauveA } from "./dark/mauveA.native.js";
import { mintA } from "./dark/mintA.native.js";
import { oliveA } from "./dark/oliveA.native.js";
import { orangeA } from "./dark/orangeA.native.js";
import { pinkA } from "./dark/pinkA.native.js";
import { plumA } from "./dark/plumA.native.js";
import { purpleA } from "./dark/purpleA.native.js";
import { redA } from "./dark/redA.native.js";
import { sageA } from "./dark/sageA.native.js";
import { sandA } from "./dark/sandA.native.js";
import { skyA } from "./dark/skyA.native.js";
import { slateA } from "./dark/slateA.native.js";
import { tealA } from "./dark/tealA.native.js";
import { tomatoA } from "./dark/tomatoA.native.js";
import { violetA } from "./dark/violetA.native.js";
import { yellowA } from "./dark/yellowA.native.js";
import { amber as amber2 } from "./light/amber.native.js";
import { blue as blue2 } from "./light/blue.native.js";
import { bronze as bronze2 } from "./light/bronze.native.js";
import { brown as brown2 } from "./light/brown.native.js";
import { crimson as crimson2 } from "./light/crimson.native.js";
import { cyan as cyan2 } from "./light/cyan.native.js";
import { gold as gold2 } from "./light/gold.native.js";
import { grass as grass2 } from "./light/grass.native.js";
import { gray as gray2 } from "./light/gray.native.js";
import { green as green2 } from "./light/green.native.js";
import { indigo as indigo2 } from "./light/indigo.native.js";
import { lime as lime2 } from "./light/lime.native.js";
import { mauve as mauve2 } from "./light/mauve.native.js";
import { mint as mint2 } from "./light/mint.native.js";
import { olive as olive2 } from "./light/olive.native.js";
import { orange as orange2 } from "./light/orange.native.js";
import { pink as pink2 } from "./light/pink.native.js";
import { plum as plum2 } from "./light/plum.native.js";
import { purple as purple2 } from "./light/purple.native.js";
import { red as red2 } from "./light/red.native.js";
import { sage as sage2 } from "./light/sage.native.js";
import { sand as sand2 } from "./light/sand.native.js";
import { sky as sky2 } from "./light/sky.native.js";
import { slate as slate2 } from "./light/slate.native.js";
import { teal as teal2 } from "./light/teal.native.js";
import { tomato as tomato2 } from "./light/tomato.native.js";
import { violet as violet2 } from "./light/violet.native.js";
import { yellow as yellow2 } from "./light/yellow.native.js";
import { amberA as amberA2 } from "./light/amberA.native.js";
import { blueA as blueA2 } from "./light/blueA.native.js";
import { bronzeA as bronzeA2 } from "./light/bronzeA.native.js";
import { brownA as brownA2 } from "./light/brownA.native.js";
import { crimsonA as crimsonA2 } from "./light/crimsonA.native.js";
import { cyanA as cyanA2 } from "./light/cyanA.native.js";
import { goldA as goldA2 } from "./light/goldA.native.js";
import { grassA as grassA2 } from "./light/grassA.native.js";
import { grayA as grayA2 } from "./light/grayA.native.js";
import { greenA as greenA2 } from "./light/greenA.native.js";
import { indigoA as indigoA2 } from "./light/indigoA.native.js";
import { limeA as limeA2 } from "./light/limeA.native.js";
import { mauveA as mauveA2 } from "./light/mauveA.native.js";
import { mintA as mintA2 } from "./light/mintA.native.js";
import { oliveA as oliveA2 } from "./light/oliveA.native.js";
import { orangeA as orangeA2 } from "./light/orangeA.native.js";
import { pinkA as pinkA2 } from "./light/pinkA.native.js";
import { plumA as plumA2 } from "./light/plumA.native.js";
import { purpleA as purpleA2 } from "./light/purpleA.native.js";
import { redA as redA2 } from "./light/redA.native.js";
import { sageA as sageA2 } from "./light/sageA.native.js";
import { sandA as sandA2 } from "./light/sandA.native.js";
import { skyA as skyA2 } from "./light/skyA.native.js";
import { slateA as slateA2 } from "./light/slateA.native.js";
import { tealA as tealA2 } from "./light/tealA.native.js";
import { tomatoA as tomatoA2 } from "./light/tomatoA.native.js";
import { violetA as violetA2 } from "./light/violetA.native.js";
import { yellowA as yellowA2 } from "./light/yellowA.native.js";
import { blackA } from "./blackA.native.js";
import { whiteA } from "./whiteA.native.js";
export { amber2 as amber, amberA2 as amberA, amber as amberDark, amberA as amberDarkA, blackA, blue2 as blue, blueA2 as blueA, blue as blueDark, blueA as blueDarkA, bronze2 as bronze, bronzeA2 as bronzeA, bronze as bronzeDark, bronzeA as bronzeDarkA, brown2 as brown, brownA2 as brownA, brown as brownDark, brownA as brownDarkA, crimson2 as crimson, crimsonA2 as crimsonA, crimson as crimsonDark, crimsonA as crimsonDarkA, cyan2 as cyan, cyanA2 as cyanA, cyan as cyanDark, cyanA as cyanDarkA, gold2 as gold, goldA2 as goldA, gold as goldDark, goldA as goldDarkA, grass2 as grass, grassA2 as grassA, grass as grassDark, grassA as grassDarkA, gray2 as gray, grayA2 as grayA, gray as grayDark, grayA as grayDarkA, green2 as green, greenA2 as greenA, green as greenDark, greenA as greenDarkA, indigo2 as indigo, indigoA2 as indigoA, indigo as indigoDark, indigoA as indigoDarkA, lime2 as lime, limeA2 as limeA, lime as limeDark, limeA as limeDarkA, mauve2 as mauve, mauveA2 as mauveA, mauve as mauveDark, mauveA as mauveDarkA, mint2 as mint, mintA2 as mintA, mint as mintDark, mintA as mintDarkA, olive2 as olive, oliveA2 as oliveA, olive as oliveDark, oliveA as oliveDarkA, orange2 as orange, orangeA2 as orangeA, orange as orangeDark, orangeA as orangeDarkA, pink2 as pink, pinkA2 as pinkA, pink as pinkDark, pinkA as pinkDarkA, plum2 as plum, plumA2 as plumA, plum as plumDark, plumA as plumDarkA, purple2 as purple, purpleA2 as purpleA, purple as purpleDark, purpleA as purpleDarkA, red2 as red, redA2 as redA, red as redDark, redA as redDarkA, sage2 as sage, sageA2 as sageA, sage as sageDark, sageA as sageDarkA, sand2 as sand, sandA2 as sandA, sand as sandDark, sandA as sandDarkA, sky2 as sky, skyA2 as skyA, sky as skyDark, skyA as skyDarkA, slate2 as slate, slateA2 as slateA, slate as slateDark, slateA as slateDarkA, teal2 as teal, tealA2 as tealA, teal as tealDark, tealA as tealDarkA, tomato2 as tomato, tomatoA2 as tomatoA, tomato as tomatoDark, tomatoA as tomatoDarkA, violet2 as violet, violetA2 as violetA, violet as violetDark, violetA as violetDarkA, whiteA, yellow2 as yellow, yellowA2 as yellowA, yellow as yellowDark, yellowA as yellowDarkA };
