// @tamagui/helpers/dist/esm/concatClassName.native.js
function concatClassName(_cn) {
  for (var args = arguments, usedPrefixes = [], final = "", len = args.length, propObjects = null, x = len; x >= 0; x--) {
    var _loop = function(i2) {
      var name = names[i2];
      if (!name || name === " ") return "continue";
      if (name[0] !== "_")
        return final = name + " " + final, "continue";
      var splitIndex = name.indexOf("-");
      if (splitIndex < 1)
        return final = name + " " + final, "continue";
      var nextChar = name[splitIndex + 1], isMediaQuery = nextChar === "_", styleKey = name.slice(1, name.lastIndexOf("-")), mediaKey = isMediaQuery ? name.slice(splitIndex + 2, splitIndex + 7) : null, uid = mediaKey ? styleKey + mediaKey : styleKey;
      if (usedPrefixes.indexOf(uid) > -1)
        return "continue";
      usedPrefixes.push(uid);
      var propName = styleKey;
      if (propName && propObjects && propObjects.some(function(po) {
        if (mediaKey) {
          var propKey = pseudoInvert[mediaKey];
          return po && po[propKey] && propName in po[propKey] && po[propKey] !== null;
        }
        var res = po && propName in po && po[propName] !== null;
        return res;
      }))
        return "continue";
      final = name + " " + final;
    }, cns = args[x];
    if (cns) {
      if (!Array.isArray(cns) && typeof cns != "string") {
        propObjects = propObjects || [], propObjects.push(cns);
        continue;
      }
      for (var names = Array.isArray(cns) ? cns : cns.split(" "), numNames = names.length, i = numNames - 1; i >= 0; i--) _loop(i);
    }
  }
  return final;
}
var pseudoInvert = {
  hover: "hoverStyle",
  focus: "focusStyle",
  press: "pressStyle",
  focusVisible: "focusVisibleStyle",
  focusWithin: "focusWithinStyle",
  disabled: "disabledStyle"
};
export {
  concatClassName
};
