// @tamagui/colors/dist/cjs/dark/amberA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var amberA_exports = {};
__export(amberA_exports, {
  amberA: () => amberA
});
module.exports = __toCommonJS(amberA_exports);
const amberA = {
  amberA1: "hsla(0, 0%, 0%, 0)",
  amberA2: "hsla(31, 100%, 49.7%, 0.036)",
  amberA3: "hsla(27, 100%, 49.9%, 0.094)",
  amberA4: "hsla(29, 100%, 50.0%, 0.143)",
  amberA5: "hsla(31, 100%, 50.0%, 0.192)",
  amberA6: "hsla(35, 100%, 50.0%, 0.250)",
  amberA7: "hsla(34, 99.6%, 52.9%, 0.331)",
  amberA8: "hsla(36, 100%, 50.0%, 0.442)",
  amberA9: "hsla(40, 100%, 57.2%, 0.980)",
  amberA10: "hsla(44, 100%, 64.2%, 0.980)",
  amberA11: "hsla(39, 99.9%, 52.7%, 0.938)",
  amberA12: "hsla(45, 100%, 94.2%, 0.980)"
};
