// @tamagui/colors/dist/cjs/dark/bronzeA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var bronzeA_exports = {};
__export(bronzeA_exports, {
  bronzeA: () => bronzeA
});
module.exports = __toCommonJS(bronzeA_exports);
const bronzeA = {
  bronzeA1: "hsla(0, 0%, 0%, 0)",
  bronzeA2: "hsla(20, 88.2%, 74.2%, 0.027)",
  bronzeA3: "hsla(10, 99.4%, 83.0%, 0.074)",
  bronzeA4: "hsla(18, 96.0%, 81.1%, 0.114)",
  bronzeA5: "hsla(18, 99.4%, 81.7%, 0.148)",
  bronzeA6: "hsla(15, 98.1%, 82.4%, 0.192)",
  bronzeA7: "hsla(16, 99.2%, 82.9%, 0.270)",
  bronzeA8: "hsla(18, 99.5%, 82.6%, 0.396)",
  bronzeA9: "hsla(18, 99.3%, 85.0%, 0.592)",
  bronzeA10: "hsla(18, 99.6%, 85.2%, 0.657)",
  bronzeA11: "hsla(17, 99.9%, 86.1%, 0.774)",
  bronzeA12: "hsla(20, 99.8%, 96.4%, 0.974)"
};
