// @tamagui/colors/dist/cjs/dark/brown.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var brown_exports = {};
__export(brown_exports, {
  brown: () => brown
});
module.exports = __toCommonJS(brown_exports);
const brown = {
  brown1: "hsl(22, 15.0%, 8.7%)",
  brown2: "hsl(20, 28.3%, 10.4%)",
  brown3: "hsl(20, 28.0%, 14.0%)",
  brown4: "hsl(21, 28.4%, 16.5%)",
  brown5: "hsl(22, 28.7%, 18.9%)",
  brown6: "hsl(23, 29.0%, 22.3%)",
  brown7: "hsl(25, 29.5%, 27.8%)",
  brown8: "hsl(27, 30.1%, 35.9%)",
  brown9: "hsl(28, 34.0%, 51.0%)",
  brown10: "hsl(28, 41.4%, 55.8%)",
  brown11: "hsl(28, 60.0%, 64.5%)",
  brown12: "hsl(30, 67.0%, 94.0%)"
};
