// @tamagui/colors/dist/cjs/dark/brownA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var brownA_exports = {};
__export(brownA_exports, {
  brownA: () => brownA
});
module.exports = __toCommonJS(brownA_exports);
const brownA = {
  brownA1: "hsla(0, 0%, 0%, 0)",
  brownA2: "hsla(22, 99.6%, 53.6%, 0.035)",
  brownA3: "hsla(18, 97.8%, 69.0%, 0.088)",
  brownA4: "hsla(21, 98.2%, 71.0%, 0.123)",
  brownA5: "hsla(25, 98.4%, 72.1%, 0.158)",
  brownA6: "hsla(25, 98.7%, 73.5%, 0.206)",
  brownA7: "hsla(25, 99.0%, 74.6%, 0.289)",
  brownA8: "hsla(28, 99.2%, 75.3%, 0.407)",
  brownA9: "hsla(28, 100%, 74.8%, 0.642)",
  brownA10: "hsla(28, 99.9%, 74.9%, 0.712)",
  brownA11: "hsla(28, 99.9%, 74.9%, 0.843)",
  brownA12: "hsla(32, 98.2%, 95.7%, 0.979)"
};
