// @tamagui/colors/dist/cjs/dark/gold.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var gold_exports = {};
__export(gold_exports, {
  gold: () => gold
});
module.exports = __toCommonJS(gold_exports);
const gold = {
  gold1: "hsl(44, 9.0%, 8.3%)",
  gold2: "hsl(43, 14.3%, 9.6%)",
  gold3: "hsl(42, 15.5%, 13.0%)",
  gold4: "hsl(41, 16.4%, 15.6%)",
  gold5: "hsl(41, 16.9%, 17.8%)",
  gold6: "hsl(40, 17.6%, 20.8%)",
  gold7: "hsl(38, 18.5%, 26.4%)",
  gold8: "hsl(36, 19.6%, 35.1%)",
  gold9: "hsl(36, 20.0%, 49.5%)",
  gold10: "hsl(36, 22.3%, 54.5%)",
  gold11: "hsl(35, 30.0%, 64.0%)",
  gold12: "hsl(49, 52.0%, 93.8%)"
};
