// @tamagui/colors/dist/cjs/dark/goldA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var goldA_exports = {};
__export(goldA_exports, {
  goldA: () => goldA
});
module.exports = __toCommonJS(goldA_exports);
const goldA = {
  goldA1: "hsla(0, 0%, 0%, 0)",
  goldA2: "hsla(40, 93.7%, 70.6%, 0.022)",
  goldA3: "hsla(40, 97.5%, 80.6%, 0.065)",
  goldA4: "hsla(40, 95.9%, 80.8%, 0.100)",
  goldA5: "hsla(38, 97.3%, 82.1%, 0.130)",
  goldA6: "hsla(39, 97.2%, 82.5%, 0.169)",
  goldA7: "hsla(37, 99.3%, 82.4%, 0.246)",
  goldA8: "hsla(35, 98.7%, 82.2%, 0.363)",
  goldA9: "hsla(36, 99.7%, 82.8%, 0.552)",
  goldA10: "hsla(35, 99.2%, 83.7%, 0.613)",
  goldA11: "hsla(35, 99.3%, 85.3%, 0.725)",
  goldA12: "hsla(49, 98.6%, 96.7%, 0.966)"
};
