// @tamagui/colors/dist/cjs/dark/grass.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var grass_exports = {};
__export(grass_exports, {
  grass: () => grass
});
module.exports = __toCommonJS(grass_exports);
const grass = {
  grass1: "hsl(146, 30.0%, 7.4%)",
  grass2: "hsl(136, 33.3%, 8.8%)",
  grass3: "hsl(137, 36.0%, 11.4%)",
  grass4: "hsl(137, 37.6%, 13.7%)",
  grass5: "hsl(136, 38.7%, 16.0%)",
  grass6: "hsl(135, 39.6%, 19.1%)",
  grass7: "hsl(134, 40.3%, 23.8%)",
  grass8: "hsl(131, 40.1%, 30.8%)",
  grass9: "hsl(131, 41.0%, 46.5%)",
  grass10: "hsl(131, 39.0%, 52.1%)",
  grass11: "hsl(131, 43.0%, 57.2%)",
  grass12: "hsl(137, 72.0%, 94.0%)"
};
