// @tamagui/colors/dist/cjs/dark/grayA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var grayA_exports = {};
__export(grayA_exports, {
  grayA: () => grayA
});
module.exports = __toCommonJS(grayA_exports);
const grayA = {
  grayA1: "hsla(0, 0%, 100%, 0)",
  grayA2: "hsla(0, 0%, 100%, 0.026)",
  grayA3: "hsla(0, 0%, 100%, 0.056)",
  grayA4: "hsla(0, 0%, 100%, 0.077)",
  grayA5: "hsla(0, 0%, 100%, 0.103)",
  grayA6: "hsla(0, 0%, 100%, 0.129)",
  grayA7: "hsla(0, 0%, 100%, 0.172)",
  grayA8: "hsla(0, 0%, 100%, 0.249)",
  grayA9: "hsla(0, 0%, 100%, 0.386)",
  grayA10: "hsla(0, 0%, 100%, 0.446)",
  grayA11: "hsla(0, 0%, 100%, 0.592)",
  grayA12: "hsla(0, 0%, 100%, 0.923)"
};
