// @tamagui/colors/dist/cjs/dark/limeA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var limeA_exports = {};
__export(limeA_exports, {
  limeA: () => limeA
});
module.exports = __toCommonJS(limeA_exports);
const limeA = {
  limeA1: "hsla(0, 0%, 0%, 0)",
  limeA2: "hsla(75, 96.4%, 59.6%, 0.022)",
  limeA3: "hsla(88, 98.0%, 70.4%, 0.061)",
  limeA4: "hsla(81, 97.8%, 67.4%, 0.096)",
  limeA5: "hsla(82, 98.4%, 65.6%, 0.135)",
  limeA6: "hsla(79, 99.7%, 64.3%, 0.182)",
  limeA7: "hsla(77, 99.1%, 62.1%, 0.252)",
  limeA8: "hsla(75, 100%, 60.0%, 0.342)",
  limeA9: "hsla(81, 99.8%, 59.7%, 0.819)",
  limeA10: "hsla(75, 99.8%, 63.7%, 0.936)",
  limeA11: "hsla(81, 99.9%, 58.7%, 0.719)",
  limeA12: "hsla(83, 100%, 94.2%, 0.980)"
};
