// @tamagui/colors/dist/cjs/dark/oliveA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var oliveA_exports = {};
__export(oliveA_exports, {
  oliveA: () => oliveA
});
module.exports = __toCommonJS(oliveA_exports);
const oliveA = {
  oliveA1: "hsla(0, 0%, 0%, 0)",
  oliveA2: "hsla(91, 97.0%, 84.0%, 0.026)",
  oliveA3: "hsla(101, 87.4%, 87.7%, 0.057)",
  oliveA4: "hsla(92, 91.8%, 94.1%, 0.078)",
  oliveA5: "hsla(101, 92.6%, 93.5%, 0.104)",
  oliveA6: "hsla(102, 91.1%, 94.6%, 0.130)",
  oliveA7: "hsla(102, 92.5%, 95.9%, 0.173)",
  oliveA8: "hsla(107, 100%, 96.5%, 0.250)",
  oliveA9: "hsla(110, 98.3%, 94.1%, 0.397)",
  oliveA10: "hsla(109, 99.6%, 95.3%, 0.457)",
  oliveA11: "hsla(113, 95.3%, 97.2%, 0.600)",
  oliveA12: "hsla(120, 93.5%, 99.6%, 0.927)"
};
