// @tamagui/colors/dist/cjs/dark/plum.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var plum_exports = {};
__export(plum_exports, {
  plum: () => plum
});
module.exports = __toCommonJS(plum_exports);
const plum = {
  plum1: "hsl(301, 20.0%, 9.4%)",
  plum2: "hsl(300, 29.8%, 11.2%)",
  plum3: "hsl(298, 34.4%, 15.3%)",
  plum4: "hsl(297, 36.8%, 18.3%)",
  plum5: "hsl(296, 38.5%, 21.1%)",
  plum6: "hsl(295, 40.4%, 24.7%)",
  plum7: "hsl(294, 42.7%, 30.6%)",
  plum8: "hsl(292, 45.1%, 40.0%)",
  plum9: "hsl(292, 45.0%, 51.0%)",
  plum10: "hsl(295, 50.0%, 55.4%)",
  plum11: "hsl(300, 60.0%, 62.0%)",
  plum12: "hsl(296, 74.0%, 95.7%)"
};
