// @tamagui/colors/dist/cjs/dark/purpleA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var purpleA_exports = {};
__export(purpleA_exports, {
  purpleA: () => purpleA
});
module.exports = __toCommonJS(purpleA_exports);
const purpleA = {
  purpleA1: "hsla(0, 0%, 0%, 0)",
  purpleA2: "hsla(280, 96.5%, 57.5%, 0.045)",
  purpleA3: "hsla(279, 98.7%, 62.8%, 0.129)",
  purpleA4: "hsla(279, 99.1%, 64.0%, 0.191)",
  purpleA5: "hsla(278, 99.8%, 64.2%, 0.248)",
  purpleA6: "hsla(276, 99.6%, 64.6%, 0.328)",
  purpleA7: "hsla(274, 99.9%, 64.6%, 0.456)",
  purpleA8: "hsla(272, 99.7%, 64.6%, 0.660)",
  purpleA9: "hsla(272, 99.9%, 69.1%, 0.748)",
  purpleA10: "hsla(273, 100%, 71.3%, 0.801)",
  purpleA11: "hsla(275, 99.9%, 75.3%, 0.934)",
  purpleA12: "hsla(286, 100%, 97.1%, 0.980)"
};
