// @tamagui/colors/dist/cjs/dark/redA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var redA_exports = {};
__export(redA_exports, {
  redA: () => redA
});
module.exports = __toCommonJS(redA_exports);
const redA = {
  redA1: "hsla(0, 0%, 0%, 0)",
  redA2: "hsla(5, 98.5%, 53.8%, 0.045)",
  redA3: "hsla(359, 99.1%, 61.1%, 0.130)",
  redA4: "hsla(358, 98.8%, 61.0%, 0.184)",
  redA5: "hsla(357, 99.6%, 60.3%, 0.237)",
  redA6: "hsla(358, 99.6%, 60.3%, 0.322)",
  redA7: "hsla(357, 100%, 59.5%, 0.442)",
  redA8: "hsla(358, 99.8%, 59.1%, 0.621)",
  redA9: "hsla(358, 100%, 65.5%, 0.884)",
  redA10: "hsla(358, 100%, 67.5%, 0.942)",
  redA11: "hsla(358, 100%, 69.7%, 0.980)",
  redA12: "hsla(352, 100%, 97.1%, 0.980)"
};
