// @tamagui/colors/dist/cjs/dark/sage.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var sage_exports = {};
__export(sage_exports, {
  sage: () => sage
});
module.exports = __toCommonJS(sage_exports);
const sage = {
  sage1: "hsl(155, 7.0%, 8.4%)",
  sage2: "hsl(150, 7.4%, 10.6%)",
  sage3: "hsl(150, 6.7%, 13.1%)",
  sage4: "hsl(150, 6.4%, 15.3%)",
  sage5: "hsl(150, 6.1%, 17.4%)",
  sage6: "hsl(150, 5.8%, 19.9%)",
  sage7: "hsl(150, 5.5%, 23.6%)",
  sage8: "hsl(150, 5.1%, 30.6%)",
  sage9: "hsl(155, 6.0%, 42.5%)",
  sage10: "hsl(153, 4.8%, 48.2%)",
  sage11: "hsl(155, 5.0%, 61.8%)",
  sage12: "hsl(155, 6.0%, 93.0%)"
};
