// @tamagui/colors/dist/cjs/dark/sageA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var sageA_exports = {};
__export(sageA_exports, {
  sageA: () => sageA
});
module.exports = __toCommonJS(sageA_exports);
const sageA = {
  sageA1: "hsla(0, 0%, 0%, 0)",
  sageA2: "hsla(123, 94.4%, 91.4%, 0.026)",
  sageA3: "hsla(123, 82.9%, 91.0%, 0.057)",
  sageA4: "hsla(124, 97.9%, 94.5%, 0.082)",
  sageA5: "hsla(125, 90.0%, 95.2%, 0.104)",
  sageA6: "hsla(142, 95.1%, 94.8%, 0.134)",
  sageA7: "hsla(143, 92.8%, 95.7%, 0.173)",
  sageA8: "hsla(146, 94.7%, 95.3%, 0.255)",
  sageA9: "hsla(151, 98.2%, 94.4%, 0.397)",
  sageA10: "hsla(148, 99.5%, 95.5%, 0.457)",
  sageA11: "hsla(152, 95.1%, 97.3%, 0.600)",
  sageA12: "hsla(149, 93.3%, 99.6%, 0.927)"
};
