// @tamagui/colors/dist/cjs/dark/skyA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var skyA_exports = {};
__export(skyA_exports, {
  skyA: () => skyA
});
module.exports = __toCommonJS(skyA_exports);
const skyA = {
  skyA1: "hsla(0, 0%, 0%, 0)",
  skyA2: "hsla(208, 100%, 49.8%, 0.045)",
  skyA3: "hsla(201, 100%, 49.8%, 0.099)",
  skyA4: "hsla(201, 100%, 50.0%, 0.148)",
  skyA5: "hsla(200, 100%, 49.8%, 0.198)",
  skyA6: "hsla(199, 100%, 49.9%, 0.256)",
  skyA7: "hsla(199, 100%, 49.9%, 0.337)",
  skyA8: "hsla(199, 100%, 50.0%, 0.453)",
  skyA9: "hsla(192, 100%, 70.8%, 0.980)",
  skyA10: "hsla(190, 100%, 77.6%, 0.980)",
  skyA11: "hsla(192, 99.9%, 59.6%, 0.924)",
  skyA12: "hsla(189, 100%, 96.8%, 0.980)"
};
