// @tamagui/colors/dist/cjs/dark/slateA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var slateA_exports = {};
__export(slateA_exports, {
  slateA: () => slateA
});
module.exports = __toCommonJS(slateA_exports);
const slateA = {
  slateA1: "hsla(0, 0%, 0%, 0)",
  slateA2: "hsla(181, 98.9%, 91.8%, 0.026)",
  slateA3: "hsla(182, 86.7%, 91.4%, 0.057)",
  slateA4: "hsla(209, 86.7%, 93.9%, 0.083)",
  slateA5: "hsla(200, 90.3%, 93.4%, 0.109)",
  slateA6: "hsla(209, 95.3%, 93.5%, 0.139)",
  slateA7: "hsla(204, 98.5%, 93.9%, 0.182)",
  slateA8: "hsla(209, 94.0%, 94.7%, 0.265)",
  slateA9: "hsla(207, 97.3%, 94.0%, 0.412)",
  slateA10: "hsla(209, 99.4%, 95.2%, 0.472)",
  slateA11: "hsla(208, 98.7%, 96.8%, 0.615)",
  slateA12: "hsla(211, 86.7%, 99.6%, 0.927)"
};
