// @tamagui/colors/dist/cjs/dark/tomatoA.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var tomatoA_exports = {};
__export(tomatoA_exports, {
  tomatoA: () => tomatoA
});
module.exports = __toCommonJS(tomatoA_exports);
const tomatoA = {
  tomatoA1: "hsla(0, 0%, 0%, 0)",
  tomatoA2: "hsla(5, 100%, 49.6%, 0.058)",
  tomatoA3: "hsla(6, 99.6%, 54.9%, 0.133)",
  tomatoA4: "hsla(6, 99.2%, 55.4%, 0.191)",
  tomatoA5: "hsla(6, 99.5%, 55.8%, 0.244)",
  tomatoA6: "hsla(7, 99.7%, 55.9%, 0.319)",
  tomatoA7: "hsla(8, 99.8%, 54.8%, 0.434)",
  tomatoA8: "hsla(10, 99.8%, 53.5%, 0.598)",
  tomatoA9: "hsla(10, 100%, 59.7%, 0.885)",
  tomatoA10: "hsla(10, 100%, 63.6%, 0.916)",
  tomatoA11: "hsla(10, 99.7%, 66.4%, 0.939)",
  tomatoA12: "hsla(12, 100%, 97.1%, 0.980)"
};
