// @tamagui/colors/dist/cjs/dark/yellowA.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var yellowA_exports = {};
__export(yellowA_exports, {
  yellowA: () => yellowA
});
module.exports = __toCommonJS(yellowA_exports);
const yellowA = {
  yellowA1: "hsla(0, 0%, 0%, 0)",
  yellowA2: "hsla(49, 100%, 49.1%, 0.027)",
  yellowA3: "hsla(45, 100%, 49.7%, 0.071)",
  yellowA4: "hsla(46, 100%, 49.7%, 0.111)",
  yellowA5: "hsla(47, 100%, 49.9%, 0.150)",
  yellowA6: "hsla(51, 100%, 49.8%, 0.199)",
  yellowA7: "hsla(51, 99.8%, 53.6%, 0.269)",
  yellowA8: "hsla(51, 100%, 49.9%, 0.371)",
  yellowA9: "hsla(53, 100%, 52.0%, 0.956)",
  yellowA10: "hsla(56, 100%, 68.4%, 0.980)",
  yellowA11: "hsla(48, 100%, 50.0%, 0.934)",
  yellowA12: "hsla(60, 100%, 91.8%, 0.980)"
};
