// @tamagui/core/dist/cjs/getBaseViews.native.js
"use strict";

var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var getBaseViews_native_exports = {};
__export(getBaseViews_native_exports, {
  getBaseViews: () => getBaseViews
});
module.exports = __toCommonJS(getBaseViews_native_exports);
function getBaseViews() {
  var _native_default,
    _native_default1,
    _native_default2,
    _native_default3,
    native = require("react-native");
  return {
    View: native.View || ((_native_default = native.default) === null || _native_default === void 0 ? void 0 : _native_default.View),
    Text: native.Text || ((_native_default1 = native.default) === null || _native_default1 === void 0 ? void 0 : _native_default1.Text),
    TextAncestor: native.unstable_TextAncestorContext,
    StyleSheet: native.StyleSheet || ((_native_default2 = native.default) === null || _native_default2 === void 0 ? void 0 : _native_default2.StyleSheet),
    Pressable: native.Pressable || ((_native_default3 = native.default) === null || _native_default3 === void 0 ? void 0 : _native_default3.Pressable)
  };
}
