// @tamagui/theme-builder/dist/cjs/index.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  },
  __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var index_exports = {};
__export(index_exports, {
  PALETTE_BACKGROUND_OFFSET: () => import_getThemeSuitePalettes.PALETTE_BACKGROUND_OFFSET,
  createPalettes: () => import_createThemes.createPalettes,
  createStudioThemes: () => import_createStudioThemes.createStudioThemes,
  createThemes: () => import_createThemes.createThemes,
  defaultComponentThemes: () => import_defaultComponentThemes.defaultComponentThemes,
  defaultTemplates: () => import_defaultTemplates.defaultTemplates,
  getThemeSuitePalettes: () => import_getThemeSuitePalettes.getThemeSuitePalettes,
  masks: () => import_masks.masks
});
module.exports = __toCommonJS(index_exports);
__reExport(index_exports, require("./ThemeBuilder.cjs"), module.exports);
__reExport(index_exports, require("@tamagui/create-theme"), module.exports);
var import_createStudioThemes = require("./createStudioThemes.cjs"),
  import_createThemes = require("./createThemes.cjs"),
  import_defaultTemplates = require("./defaultTemplates.cjs"),
  import_defaultComponentThemes = require("./defaultComponentThemes.cjs"),
  import_getThemeSuitePalettes = require("./getThemeSuitePalettes.cjs"),
  import_masks = require("./masks.cjs");
